corehq/apps/hqwebapp/templates/hqwebapp/base_navigation.html

```diff
@@ -38,7 +38,7 @@
         <nav class="navbar-menus navbar-signin" role="navigation">
           <div class="nav-settings-bar pull-right">
             <a href="{% url "login" %}" class="btn btn-primary navbar-btn">{% trans 'Sign In' %}</a>
-            {% if is_saas_environment and ANALYTICS_IDS.HUBSPOT_API_ID %}
+            {% if is_saas_environment and ANALYTICS_IDS.HUBSPOT_API_ID and is_hubspot_js_allowed %}
```

Comment:
Since the context processor is already setting `is_hubspot_js_allowed` to `False` if this isn't a SAAS environment, would it make sense to remove the `is_saas_environment` clause?